src/SqliteSampleBlock.cpp

```diff
@@ -1008,6 +1011,57 @@ void SqliteSampleBlock::CalcSummary(Sizes sizes)
    mSumMax = max;
 }
 
+//! Just to find a denominator for a progress indicator.
+/*! This estimate procedure should in fact be exact */
+static size_t EstimateRemovedBlocks(
+   AudacityProject &project, size_t begin, size_t end)
+{
+   auto &manager = UndoManager::Get(project);
+
+   // Collect ids that survive
+   SampleBlockIDSet wontDelete;
+   auto f = [&](const UndoStackElem &elem){
+      InspectBlocks(*elem.state.tracks, {}, &wontDelete);
+   };
+   manager.VisitStates(f, 0, begin);
+   manager.VisitStates(f, end, manager.GetNumStates());
+   if (const auto saved = manager.GetSavedState(); saved >= 0)
+      manager.VisitStates(f, saved, saved + 1);
+   InspectBlocks(TrackList::Get(project), {}, &wontDelete);
+
+   // Collect ids that won't survive (and are not negative pseudo ids)
+   SampleBlockIDSet seen, mayDelete;
+   manager.VisitStates( [&](const UndoStackElem &elem){
+      auto &tracks = *elem.state.tracks;
+      InspectBlocks(tracks, [&]( const SampleBlock &block ){
+         auto id = block.GetBlockID();
+         if ( id > 0 && !wontDelete.count( id ) )
+            mayDelete.insert( id );
+      },
+      &seen);
+   }, begin, end );
+   return mayDelete.size();
+}
+
+void SqliteSampleBlockFactory::OnBeginPurge(size_t begin, size_t end)
+{
+   // Install a callback function that updates a progress indicator
+   using namespace BasicUI;
+   auto pDialog = std::shared_ptr<ProgressDialog>{ MakeProgress(
```

Comment:
As discuss, this fails to build when running under Visual Studio 2019 IDE.

Had to change to std::shared_ptr<BasicUI::ProgressDialog> for it to compile.

Reply:
Removing the ProgressDialog.h, from AudacityHeaders.h did the trick.

Reply:
Removing the ProgressDialog.h, from AudacityHeaders.h did the trick.